addons/mrp/report/stock_forecasted.py

```diff
@@ -57,3 +57,21 @@ def _get_reservation_data(self, move):
             'name': move[m2o].name,
             'id': move[m2o].id
         }
+
+    @api.model
+    def action_reserve_linked_picks(self, move_id):
+        """ In case of move connected to MOs instead of picking we want to reserve the MO
+        """
+        move_id = self.env['stock.move'].browse(move_id)
+        linked_move_ids = move_id.browse(move_id._rollup_move_origs())
```

Comment:
The `_rollup_move_origs` part is the same than in the super, it would be better to avoid to do it two times since we only want to check it there is `production_id` or `raw_material_production_id` on the exact same moves found by the super

Reply:
I could call the super first and have it return the `move_ids` to be reused :thinking: